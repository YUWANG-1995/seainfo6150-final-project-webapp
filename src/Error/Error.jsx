import React from 'react'
import error from "../images/Error/error.png";
import styles from "./Error.module.css";
import { Link } from "react-router-dom";


const Error = () => {
    return (
        <div className={styles.container}>
          <div>
            <p className={styles.text}>
              404 Not Found !<br />
              Ooops, here is something wrong.<br/>
              <Link to="/"><span>Click me back to home page</span></Link>
            </p>
          </div>
          <div>
            <img src={error} alt="404 Not Found" />
          </div>
        </div>
    )
}

export default Error
