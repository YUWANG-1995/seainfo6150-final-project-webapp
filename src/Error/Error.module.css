.container{
  display: grid;
  grid-auto-rows: 1fr 4fr;
  justify-content: center;
  align-items: center;
}

.text{
  font-family: 'Open Sans', sans-serif;
  font-size: 1.5em;
  font-weight: bolder;
  text-align: center;
  font-style: italic;
}
