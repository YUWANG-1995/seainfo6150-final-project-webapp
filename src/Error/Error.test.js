import React from "react";
import { render } from "@testing-library/react";
import Error from "./Error.jsx";
import { BrowserRouter } from 'react-router-dom'

it("renders the Error Component Correctly", () => {
  const { container } = render(
    <BrowserRouter>
      <Error />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
