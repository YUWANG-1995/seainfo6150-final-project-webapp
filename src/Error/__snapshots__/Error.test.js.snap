// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Error Component Correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <p
        class="text"
      >
        404 Not Found !
        <br />
        Ooops, here is something wrong.
        <br />
        <a
          href="/"
        >
          <span>
            Click me back to home page
          </span>
        </a>
      </p>
    </div>
    <div>
      <img
        alt="404 Not Found"
        src="error.png"
      />
    </div>
  </div>
</div>
`;
